code/modules/events/grid_check.dm

```diff
@@ -67,6 +72,11 @@
 		S.input_attempt = S.last_input_attempt
 		S.update_icon()
 		S.power_change()
+	for(var/obj/machinery/power/P in GLOB.machines)
+		var/area/current_area = get_area(P)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(P.z))
+			continue
+		P.malfunction = FALSE
```

Comment:
Bit of a nitpick, but if I have a plasma gen or something on the station level and it malfunctions, I can then take it across z levels and it won't ever stop malfunctioning since it will be skipped when the event ends.